### PR TITLE
Glitch

### PR DESCRIPTION
**Pull request by sciencefixion**
### New Feature Submissions:
* [X] Have you added an explanation of what your changes do?
* [X] Have you linted your code locally prior to submission?

### Testing:
* [?] Have you written new tests for your core changes, as applicable?
* [X] Have you successfully ran tests with your changes locally?
* [X] All new and existing tests passed?

**Test coverage** = 99.02%
--------------------------------------------------------------------------------
### Change Log:

*Changed Image class to ApiImage to avoid issues with ImageMagick's Image class
*Added pnglitch and mini_magick gems
*Added Glitch functionality with the Glitchable module
*Added glitched attachment to ApiImage class and saved to ActiveStorage
*One skipped test and commented route that still needs fixing

*Rebundle and migrate

@janegreene @dtsehl @Kathybui732

### CODE REVIEW

**Comment by sciencefixion** [comment]
*Note @janegreene @dtsehl @Kathybui732 all assisted with this branch even though commits do not reflect that. My bad!

File: app/controllers/search_controller.rb [hunk 1]
```diff
@@ -4,18 +4,24 @@ def index
     search = params[:search].split.join('+')
     if search.match('[a-zA-Z]+')
       conn = Faraday.get("https://damp-forest-93176.herokuapp.com/images/#{search}")
-      if conn.status == 500
-        redirect_to '/500'
-      elsif conn.status == 200
-        json = JSON.parse(conn.body, symbolize_names: true)
-        @image_options = ImageOptions.new(json)
-      end
+      get_image_options(conn)
```
Comment (dtsehl) [inline app/controllers/search_controller.rb:7]: Nice going with this method! Makes the controller action much slimmer.

File: app/controllers/search_controller.rb [hunk 2]
```diff
@@ -4,18 +4,24 @@ def index
...
     end
   end
 
   def show
-    @image = Image.find(params[:id])
+    @image = ApiImage.find(params[:id])
+  end
+
+  private
+
```
Comment (janegreene) [inline app/controllers/search_controller.rb:18]: Nice refactor, thanks for pleasing the rubocop!

File: app/models/glitchable.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+require 'pnglitch'
+require 'pry'
+
+module Glitchable
+  def convert_to_png(image)
+    target = MiniMagick::Image.open(image.url)
+    target.format 'png'
+  end
+
+  def basic_glitch(image)
```
Comment (dtsehl) [inline app/models/glitchable.rb:11]: Next step: Less basic glitch!

File: app/views/search/index.html.erb [hunk 4]
```diff
@@ -1,5 +1,52 @@
...
+  <!-- The video -->
+    <video autoplay muted loop id="myVideo">
+      <source src="https://i.imgur.com/mlcMvUl.mp4" type="video/mp4">
+  </video>
 
-<iframe src=<%= @image_options.pixabay_url %> width="645px" height="395px"></iframe>
-<%= button_to "Select Pixabay Image", image_path, method: :post, params: {url: @image_options.pixabay_url, source: @image_options.pixabay_source, deep_id: @image_options.pixabay_id} %>
+<!-- Optional: some overlay text to describe the video -->
+  <div class="content">
+    <h1>Heading</h1>
```
Comment (janegreene) [inline app/views/search/index.html.erb:9]: We can add some user instructions here to help them out.

File: config/routes.rb [hunk 5]
```diff
@@ -11,5 +11,5 @@
   post '/image', to: 'images#create'
   get '/500', to: 'application#server_errors'
   get '/400', to: 'application#search_errors'
-  get '(*url)', to: redirect('/404')
+  # get '(*url)', to: redirect('/404')
```
Comment (dtsehl) [inline config/routes.rb:14]: I wonder how to get this to work eventually.

Comment (kathytbui) [reply config/routes.rb:14]: This one line held us up all yesterday - so unfortunate

**Review by janegreene — APPROVED** [review]

**Pull request merged**